crates/aptos/src/common/types.rs

```diff
@@ -147,20 +151,38 @@ impl Default for CliConfig {
 
 impl CliConfig {
     /// Checks if the config exists in the current working directory
-    pub fn config_exists() -> CliTypedResult<bool> {
-        Self::aptos_folder().map(|folder| folder.exists())
+    pub fn config_exists() -> bool {
+        if let Ok(folder) = Self::aptos_folder() {
+            let config_file = folder.join(CONFIG_FILE);
+            let old_config_file = folder.join(LEGACY_CONFIG_FILE);
+            config_file.exists() || old_config_file.exists()
+        } else {
+            false
+        }
     }
 
     /// Loads the config from the current working directory
     pub fn load() -> CliTypedResult<Self> {
-        let config_file = Self::aptos_folder()?.join("config.yaml");
-        if !config_file.exists() {
-            return Err(CliError::ConfigNotFoundError(format!("{:?}", config_file)));
+        let folder = Self::aptos_folder()?;
+
+        let config_file = folder.join(CONFIG_FILE);
+        let old_config_file = folder.join(LEGACY_CONFIG_FILE);
```

Comment:
nit: Can you do config_file = config_file.or_else(|| old_config_file) for less redundancy?

Reply:
You have to check the files independently here.  So, or else it won't be the same.